### PR TITLE
Project admin add to cohort

### PR DESCRIPTION
**Pull request by emcelroy**

### CODE REVIEW

**Comment by emcelroy** [comment]
@scytacki: As I understand it, the add_teachers_to_cohorts method I added was always returning true when the logged in user was an admin. Since

**Comment by emcelroy** [comment]
@scytacki: I think I may be a little mixed up with how this is all working, so this suggested change may not be the thing to do. As I understand it, and as it appeared to function in my local dev portal, the `add_teachers_to_cohorts` method I originally added to the user policy was always returning true when the logged in user was an admin. Since the policy was used to determine whether any user being edited had their "User can add teachers to cohorts..." option checked, it meant every user had that option checked by default regardless of what the related value in the database was set to.

I changed it so the method checks `record` instead of `user` and that fixed the problem. However, it's not clear to me whether that's the correct thing to do. Could that break the use of the policy elsewhere? It seems like everything now works as expected in my dev portal, but I may be missing something. Maybe we should be using some other way to determine if that checkbox on the user edit form is checked?

File: app/views/users/_admin_for_projects.html.haml [hunk 1]
```diff
@@ -12,6 +12,6 @@
           = p.name
   %div.additional-options
     = hidden_field(:user, :can_add_teachers_to_cohorts, :value => false)
-    = check_box_tag "user[can_add_teachers_to_cohorts]", 1, policy(@user).add_teachers_to_cohorts?
-    = label_tag "user_can_add_teachers_to_cohorts" do
-      User can add teachers to cohorts of projects they administrate
+    = check_box_tag "user[can_add_teachers_to_cohorts]", 1, policy(@user).add_teachers_to_cohorts?, id: "user_can_add_teachers_to_cohorts_cb"
```
Comment (scytacki) [inline app/views/users/_admin_for_projects.html.haml]: Based on this documentation: `check_box_tag(name, value = "1", checked = false, options = {})` the checkbox will be 'checked' based on the value of the policy. I think that is what you wrote. 

That isn't how it should work. The value for whether it is checked should be based on value in the database for the @user record.  The `policy(@user)...` should probably not be used here. 

In this form we want to know if a admin can add this setting to another admin. We could add another policy method here. But I think it'd be fine just to check if current_user is a portal admin. We should check the controller that handles this setting to make sure non portal admins cannot set this value on another user.

File: app/policies/user_policy.rb [hunk 2]
```diff
@@ -74,7 +74,7 @@ def reset_password?
   end
 
   def add_teachers_to_cohorts?
-    (user.can_add_teachers_to_cohorts? || admin_or_manager?)
+    record.can_add_teachers_to_cohorts? || !record_not_manager? || !record_not_admin?
```
Comment (scytacki) [inline app/policies/user_policy.rb]: I think probably this should be reverted. Based on the comment below.

**Pull request merged**